discovery-infra/test_infra/controllers/iptables.py

```diff
@@ -0,0 +1,45 @@
+import logging
+
+from test_infra.utils import run_command
+
+class IptableRule:
+    def __init__(self, chain, target, protocol, dest_port='', sources=None, extra_args=''):
+        self._chain = chain
+        self._target = target
+        self._protocol = protocol
+        self._dest_port = dest_port
+        self._sources = sources if sources else []
+        self._extra_agrs = extra_args
+
+    def _build_command_string(self, option):
+        sources_string = ','.join(self._sources)
+        rule = f'iptables --{option}'
```

Comment:
This isn't so readable..  Maybe a core list with `" ".join(...)` and only add the relevant flags if needed? 
i.e.
```python
rule = ["iptables", f"--{option}", self._chain, "-p" ,self._protocol, "-j", self._target]

if self.sources:
    rule += ["-s", sources_string]

if self._dest_port:
    rule += ["--dport", self._dest_port]

if self._extra_args:
    rule += [self._extra_args]

return " ".join(rule_template)

Reply:
done